### PR TITLE
ARROW-2073: [Python] Create struct array from sequence of tuples

### PR DESCRIPTION
**Pull request by pitrou**

### CODE REVIEW

**Comment by pitrou** [comment]
AppVeyor build at https://ci.appveyor.com/project/pitrou/arrow/build/1.0.45

File: python/pyarrow/tests/test_convert_builtin.py [hunk 1]
```diff
@@ -531,6 +531,45 @@ def test_struct_from_dicts():
...
+    assert arr.to_pylist() == expected
+
+    # Invalid tuple size
+    for tup in [(5, 'foo'), (), ('5', 'foo', True, None)]:
+        with pytest.raises(ValueError, match="(?i)tuple size"):
+            pa.array([tup], type=ty)
+
+
+def test_struct_from_mixed_sequence():
+    # It is forgotten to mix dicts and tuples when initializing a struct array
```
Comment (xhochy) [inline python/pyarrow/tests/test_convert_builtin.py]: Typo: s/forgotten/forbidden/

**Review by xhochy — COMMENTED** [review]
Looks good except one small typo

**Comment by pitrou** [comment]
Ok, I fixed the typo.

**Review by xhochy — APPROVED** [review]
Thanks!

**Pull request closed without merging**